File: mobile-flutter/lib/models/pengaduan.dart
class Pengaduan {
  final int id;
  final int pelangganId;
  final String judulPengaduan;
  final String kategori;
  final String deskripsi;
  final String? tanggalPengaduan;
  final String status;
  final String? tanggapan;
  final String? tanggalTanggapan;

  Pengaduan({
    required this.id,
    required this.pelangganId,
    required this.judulPengaduan,
    required this.kategori,
    required this.deskripsi,
    this.tanggalPengaduan,
    required this.status,
    this.tanggapan,
    this.tanggalTanggapan,
  });

  factory Pengaduan.fromJson(Map<String, dynamic> json) {
    return Pengaduan(
      id: json['id'],
      pelangganId: json['pelanggan_id'],
      judulPengaduan: json['judul_pengaduan'] ?? '',
      kategori: json['kategori'] ?? '',
      deskripsi: json['deskripsi'] ?? '',
      tanggalPengaduan: json['tanggal_pengaduan'],
      status: json['status'] ?? 'Baru',
      tanggapan: json['tanggapan'],
      tanggalTanggapan: json['tanggal_tanggapan'],
    );
  }
}

File: app/Models/Pengaduan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Pengaduan extends Model
{
    use HasFactory;

    protected $table = 'pengaduan';

    protected $fillable = [
        'pelanggan_id',
        'judul_pengaduan',
        'kategori',
        'deskripsi',
        'tanggal_pengaduan',
        'status',
        'tanggapan',
        'petugas_id',
        'tanggal_tanggapan',
    ];

    protected $casts = [
        'tanggal_pengaduan' => 'datetime',
        'tanggal_tanggapan' => 'datetime',
    ];

    public function pelanggan()
    {
        return $this->belongsTo(Pelanggan::class);
    }

    public function petugas()
    {
        return $this->belongsTo(User::class, 'petugas_id');
    }
}
